### PR TITLE
[Release PR] Add StoreKit debugging menu options

### PR DESCRIPTION
**Pull request by samsymons**
Task/Issue URL: https://app.asana.com/0/1199333091098016/1208862631410451/f
Tech Design URL:
CC:

**Description**:

This PR adds debug menu entries for viewing StoreKit metadata.

**Optional E2E tests**:
- [ ] Run PIR E2E tests
	Check this to run the Personal Information Removal end to end tests. If updating CCF, or any PIR related code, tick this.

**Steps to test this PR**:
1. Open the Debug menu -> Subscription section and check that the new debugging menu options are visible

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

File: LocalPackages/SubscriptionUI/Sources/SubscriptionUI/DebugMenu/SubscriptionDebugMenu.swift [hunk 1]
```diff
@@ -92,6 +93,12 @@ public final class SubscriptionDebugMenu: NSMenuItem {
         environmentItem.submenu = makeEnvironmentSubmenu()
         menu.addItem(environmentItem)
 
+        menu.addItem(.separator())
+        let storefrontID = SKPaymentQueue.default().storefront?.identifier ?? "nil"
+        menu.addItem(NSMenuItem(title: "Storefront ID: \(storefrontID)", action: nil, target: nil))
+        let storefrontCountryCode = SKPaymentQueue.default().storefront?.countryCode ?? "nil"
```
Comment (samsymons) [inline LocalPackages/SubscriptionUI/Sources/SubscriptionUI/DebugMenu/SubscriptionDebugMenu.swift:99]: This API was deprecated in macOS 15, but it'll be a long time until we're dropping support for it so we're safe to use it here. This API is easier for this use that then new `async` option, but please let me know if you see an issue with using it.

**Comment by samsymons** [comment]
Phishing tests failed, but it looks unrelated and due to the same reason mentioned by Bartek in a couple MM channels.

File: LocalPackages/SubscriptionUI/Sources/SubscriptionUI/DebugMenu/SubscriptionDebugMenu.swift (hunk 1 above)
Comment (miasma13) [reply LocalPackages/SubscriptionUI/Sources/SubscriptionUI/DebugMenu/SubscriptionDebugMenu.swift:99]: Makes sense 👍

**Comment by samsymons** [comment]
The failed UI tests here are unrelated to this and have been known to fail on the macOS 15 runner, it's being looked at separately. Since everything else is green, I'll merge this.

**Pull request merged**